### ISSUE TITLE
Short address endianness 

### ISSUE THREAD

**Issue by tim-ist** [contiki-os/contiki#2356]
While porting Contiki to a new platform we discovered some inconsistencies in byte order of short addresses and node IDs. We tried two existing platforms, the results are in the following. In all cases, I am considering what Wireshark reports as the actually used short address. Here I use **u16** and **u8** to refer to **linkaddr_node_addr.u16** and **linkaddr_node_addr.u8** respectively.

**For srf06-cc26xx platform:**
1. Wireshark reports src address **0x3412**, with it being written in the packet as 12:34 (which is correct since the 802.15.4 headers are little-endian).
1. When I **printf("0x%x", u16)**, I am getting **0x1234**, reverse w.r.t. Wireshark.
1. However, the node_id is printed as **0x3412**, as the address in Wireshark

**For sky platform:**
1. Wireshark reports src address **0x3412** (in packet 12:34).
1. When printing **u16** I get it reverse (**0x1234**), like with the first platform.
1. node_id is printed as **0x1234**, again reverse, this time different from cc26xx.

So, the two platforms agree about the short address representation but do not about node IDs. Node IDs are not a big deal, however the address handling looks counterintuitive to me.

I would find reasonable two approaches: either **1)** store the address in memory as a 16-bit number, following the native byte order of the MCU and swap it when writing in the packets (on big-endian platforms only) or **2)** always store it in the order it should be sent over the air (little-endian). 

Contiki seems to implement neither of those at least on the two mentioned platforms.
I demonstrate it in the following examples. In both cases I want to use **0x1234** short address.

1) Following the first approach, I set **u16=0x1234**, but I get 12:34 in the air (which stands for 0x3412 address).

2) Following the second approach, I set **u8[0] = 0x34, u8[1] = 0x12** (little endian), but I still get the wrong 12:34 in the air (0x3412 address).

In other words, currently the addresses are always stored as big-endian numbers in memory, contradicting both the endianness of the 802.15.4 and the one of the MCUs. 
If it is the way it should be, it would be nice to have it documented somewhere.

By the way, PAN IDs are handled as expected (both stored and transmitted as little-endian numbers).
